### PR TITLE
Bump CAPI to v1.8.5

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
Follows: https://cluster-api.sigs.k8s.io/developer/providers/migrations/v1.7-to-v1.8

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

**Review by alexander-demicev — DISMISSED** [review]

**Review by Danil-Grigorev — DISMISSED** [review]

**Review by salasberryfin — APPROVED** [review]

File: bootstrap/main.go [hunk 1]
```diff
@@ -105,9 +107,18 @@ func InitFlags(fs *pflag.FlagSet) {
...
+	fs.IntVar(&clusterCacheTrackerConcurrency, "clustercachetracker-concurrency", 10,
+		"Number of clusters to process simultaneously")
+
 	fs.DurationVar(&syncPeriod, "sync-period", consts.DefaultSyncPeriod,
 		"The minimum interval at which watched resources are reconciled (e.g. 15m)")
 
+	fs.Float32Var(&clusterCacheTrackerClientQPS, "clustercachetracker-client-qps", 20,
+		"Maximum queries per second from the cluster cache tracker clients to the Kubernetes API server of workload clusters.")
+
+	fs.IntVar(&clusterCacheTrackerClientBurst, "clustercachetracker-client-burst", 30,
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply bootstrap/main.go:120]: thanks for noticing, should be fixed now in https://github.com/rancher/cluster-api-provider-rke2/pull/477/commits/cfbca75fca4d387b89989db479b763996774a462

**Review by Danil-Grigorev — DISMISSED** [review]

File: test/e2e/helpers.go [hunk 2]
```diff
@@ -533,3 +536,133 @@ func CollectArtifacts(ctx context.Context, kubeconfigPath, name string, args ...
 	fmt.Printf("stdout:\n%s\n", string(stdout.Bytes()))
 	return err
 }
+
+// Apply wraps `kubectl apply ...` and prints the output so we can see what gets applied to the cluster.
+func Apply(ctx context.Context, clusterProxy framework.ClusterProxy, resources []byte, args ...string) error {
```
Comment (Danil-Grigorev) [inline test/e2e/helpers.go:541]: Out of curiosity - was the usage of `Apply` what unblocked e2e?

File: bootstrap/main.go [hunk 3]
```diff
@@ -105,9 +108,18 @@ func InitFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&concurrencyNumber, "concurrency", 1,
 		"Number of core resources to process simultaneously")
 
+	fs.IntVar(&clusterCacheTrackerConcurrency, "clustercachetracker-concurrency", 10,
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply bootstrap/main.go:111]: It is not needed, I'll remove it in a follow-up PR

File: test/e2e/helpers.go (hunk 2 above)
Comment (furkatgofurov7) [reply test/e2e/helpers.go:541]: Yes

**Review by alexander-demicev — APPROVED** [review]

**Review by Danil-Grigorev — APPROVED** [review]

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**